R/gitbook.R

```diff
@@ -130,10 +130,20 @@ gitbook_page = function(
   # gitbook JS scripts only work after the DOM has been loaded, so move them
   # from head to foot
   i = grep('^\\s*<script src=".+/gitbook([^/]+)?/js/[.a-z-]+[.]js"></script>\\s*$', head)
-  # it is probably a self-contained page, so look for base64 encoded scripts
-  if (length(i) == 0) i = grep(
-    '^\\s*<script src="data:application/x-javascript;base64,[^"]+"></script>\\s*$', head
-  )
+  # it is probably a self-contained page, so look for script node.
+  # from pandoc2, they are not always base64 encoded scripts, so start and end of scripts
+  # node must be found and moved.
+  if (length(i) == 0) {
+    s_start = grep(
+      '^\\s*<script( src="data:application/(x-)?javascript;base64,[^"]+")?>',
```

Comment:
As it was before with self_contained document, with this regex, every `<script>` nodes is in fact found and moved in footer. While in not self_contained mode, only js script from gitbook are found and moved (see line 132 above). 
With this behavior, this make me think that the regex could be simplified

But I wonder if this should be improved or not... like a trick using a custom comment header in all js script to find them as node in the html document. This would allow to filter gitbook only js. 
However, I am not sure I see any harm in moving them all at the end of the body.
If we want to improve this, using another method than regex finding + moving lines would be better I guess.